sdks/go/pkg/beam/core/runtime/xlangx/expansionx/process.go

```diff
@@ -0,0 +1,67 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expansionx
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+// ExpansionServiceRunner is a type that holds information required to
+// start up a Beam Expansion Service JAR and maintain a handle on the
+// process running the service to enable shutdown as well.
+type ExpansionServiceRunner struct {
+	jarPath        string
+	serviceCommand *exec.Cmd
+}
+
+// NewExpansionServiceRunner builds an ExpansionServiceRunner struct for a given gradle target and
+// Beam version and returns a pointer to it.
+func NewExpansionServiceRunner(jarPath string) *ExpansionServiceRunner {
+	serviceCommand := exec.Command("java", "-jar", jarPath, "8097")
```

Comment:
Good future improvement: Auto-selecting an open port somehow and saving the address. In particular because this can cause errors if something's already using port 8097 (the Flink runner, for example, uses that for its own expansion service that it starts up).

Reply:
I altered the struct and factory function to take a port as an argument, which would enable adding auto-selecting an open port later.